mate-window-picker-applet/task-item.c

```diff
@@ -134,7 +134,10 @@ on_task_item_button_released (GtkWidget      *widget,
     }
     if (wnck_window_is_active (window))
```

Comment:
Better:
```
if (wnck_window_is_active (window) && !wnck_window_is_minimized (window))
{
         wnck_window_minimize (window);
}
```

Reply:
Thank you for the review, I’ll fix it.

Reply:
In fact, it is not really the same. Because the whole picture is:

```c
if (A)
    if (B)
        // do something
else
    // do otherwise
```

Meaning, the `else` close only apply to the first `if`, and not to the second. If we merge the two if in one, it would mean that the else branch will be used in wrong time.

What we want here is:

```
if window_is_active
     if window_is_not_minimized
         what we wanted when we click on the icon is to minimize it, thus minimizing it now.
     else
        window cannot be both active and minimized, thus this condition does not matter
else
    window was not active, thus what we want to do is to activate it (i.e. to raise it).
```

I’m not sure of the risk of activating an already activated window?

However you are right that the whole logic here may be improved. I’ll try to rewrite it.

Reply:
Yes you are right, I did not see the else part. Sorry.